lib/trino-orc/src/main/java/io/trino/orc/metadata/OrcMetadataWriter.java

```diff
@@ -128,13 +150,28 @@ public int writeFooter(SliceOutput output, Footer footer)
                         .map(OrcMetadataWriter::toUserMetadata)
                         .collect(toList()));
 
-        if (!useLegacyVersion) {
-            builder.setWriter(PRESTO_WRITER_ID);
-        }
+        setWriter(builder);
 
         return writeProtobufObject(output, builder.build());
     }
 
+    private void setWriter(OrcProto.Footer.Builder builder)
+    {
+        switch (writerIdentification) {
+            case LEGACY_HIVE_COMPATIBLE:
+                return;
+
+            case PRESTO:
```

Comment:
Do we want to drop this one eventually? If so maybe it should also be prefixed with `LEGACY_`

Reply:
LEGACY_HIVE_COMPATIBLE is supposed to mean "use writer identification compatible with legacy Hive versions: 2.0.x-2.2.x"

i can mark both options as deprecated if it helps. would it?

Reply:
Oh I see. No leave it. It is consistent :)